lightningd/peer_control.c

```diff
@@ -1390,6 +1409,32 @@ static struct command_result *json_setchannelfee(struct command *cmd,
 		   NULL))
 		return command_param_failed();
 
+	// If the users requested 'all' channels we need to iterate
```

Comment:
Ok, I think the better way to do this is inside a new boutique (ie static local) param_peerid_or_all: this would do the id lookup, and return pointer to peer, or NULL for all.

In general, using param_tok is an antipattern: it means that when someone runs check on this command, it accepts anything.

Reply:
Okay, good idea to get rid of the iffy stuff in the command handler.
I am still working on this one...

Reply:
Okay, also addressed 
85449b2